storage/innobase/fsp/fsp0file.cc

```diff
@@ -691,11 +697,14 @@ dberr_t Datafile::validate_first_page(space_id_t space_id, lsn_t *flush_lsn,
         ut::free(m_encryption_iv);
         m_encryption_key = nullptr;
         m_encryption_iv = nullptr;
-        xb::info() << "Failed to decrypt table " << m_filepath << " with space"
-                   << " id " << m_space_id
-                   << ". Will check if encrytion key has been"
-                   << " parsed at the end of backup.";
-        invalid_encrypted_tablespace_ids.push_back(m_space_id);
+        if (srv_backup_mode) {
```

Comment:
Nice, this also fixes the bogus message during prepare.

Reply:
there is other secret intention for doing this change.

Tablespace loading is parallel (open_ibds_slice). And each thread can insert into this invalid_encrypted_tabelsapce_ids parallel and there is no mutex protection for this structure. Avoiding a possible race condition

Reply:
Just checked code, looks like this is a problem for backup too? :thinking:

Reply:
IS the initial scan done in parallel on --backup too now ?

Reply:
yes, here is the code flow
```
Tablespace_dirs::scan->Tablespace_dirs::open_ibd->fil_open_for_xtrabackup

Tablespace_dirs::scan
  par_for(PFS_NOT_INSTRUMENTED, ibd_files, n_threads, check, &m, &unique,
            &duplicates);
  }

```

Reply:
but we have to test and see if the threads are more than 1 or not

Reply:
Can you open a separate ticket for this ?

Reply:
Done marce: https://jira.percona.com/browse/PXB-3118